src/explorer/siteNodeBase.ts

```diff
@@ -208,46 +192,53 @@ export class SiteNodeBase extends NodeBase {
         // the scm url lives in the 1 index of enabledHostNames, not 0
         const remote = `https://${username}:${password}@${repo}`;
 
-
-        let git = require('simple-git/promise')(workspace.rootPath);
-
-        try {
-            let status = await git.status();
-            if (status.files.length > 0) {
-                window.showWarningMessage(`There ${status.files.length > 1 ? 'are' : 'is'} ${status.files.length} uncommitted change${status.files.length > 1 ? 's' : ''} in local repo "${workspace.rootPath}"`);
-            }
-            await git.push(remote, 'master');
-        }
-        catch (err) {
-            if (err.message.indexOf('spawn git ENOENT') >= 0) {
-                let input = await window.showErrorMessage(`Git must be installed to use Local Git Deploy.`, `Install`)
-                if (input === 'Install') {
-                    opn(`https://git-scm.com/downloads`);
+        const git = require('simple-git/promise')(workspace.rootPath);
+        let forcePush = false;
+        for (let i = 0; i < 2; i++) {
+            try {
+                if (config.scmType !== 'LocalGit') {
+                    await this.convertScmType(this, config);
                 }
-                throw err;
-            } else if (err.message.indexOf('error: failed to push') >= 0) {
-                let input = await window.showErrorMessage(`Push rejected due to Git history diverging. Force push?`, `Yes`)
-                if (input === 'Yes') {
-                    await git.push(['-f', remote, 'master']);
+                const status = await git.status();
+                if (!forcePush && status.files.length > 0) {
+                    window.showWarningMessage(`There ${status.files.length > 1 ? 'are' : 'is'} ${status.files.length} uncommitted change${status.files.length > 1 ? 's' : ''} in local repo "${workspace.rootPath}"`);
+                }
+
+                !forcePush ?
+                    await git.push(remote, 'HEAD:master') :
+                    await git.push(['-f', remote, 'HEAD:master']);
+                break;
+            } catch (err) {
+                if (err.message.indexOf('spawn git ENOENT') >= 0) {
+                    const input = await window.showErrorMessage(`Git must be installed to use Local Git Deploy.`, `Install`);
+                    if (input === 'Install') {
+                        opn(`https://git-scm.com/downloads`);
+                    }
+                    throw err;
+                } else if (err.message.indexOf('error: failed to push') >= 0) {
+                    const input = await window.showErrorMessage(`Push rejected due to Git history diverging. Force push?`, `Yes`);
+                    if (input === 'Yes') {
+                        forcePush = true;
```

Comment:
Can we just call `await git.push(['-f', remote, 'HEAD:master']);` here and get rid of the for-loop

Reply:
When the await git.push(['-f', remote, 'HEAD:master']);  was  on line 221, there was no error handling.  I'd have to write another try/catch loop within my catch.

Reply:
What error handling, though? We know git is already installed...and you're already doing a force push so the `error: failed to push` logic doesn't apply